esmvalcore/_data_finder.py

```diff
@@ -61,7 +61,7 @@ def get_start_end_year(filename):
     #
     # Next string allows to test that there is an allowed delimiter (or
     # string start or end) close to date range (or to single date)
-    context = r"(?:^|[-_]|$)"
+    context = r"(?:^|[-_.]|$)"
```

Comment:
I'm afraid that this change would break recognizing the date range in the example that motivated reworking this function, as explained in #1127, i.e. : 
`CM61-LR-hist-03.1950_18500101_19491231_1M_concbc.nc`
Maybe introducing more constraints (e.g. length of date patterns in a range) would solve the problem ?

Reply:
Or allowing the user to provide, on a per dataset basis, a date range recognition pattern, or a date delimiter

Reply:
Converation resoved, because I had miss the further change (the one that first try without such an extended context)